pkg/console/controllers/clidownloads/controller.go

```diff
@@ -0,0 +1,295 @@
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	// informers
+	consoleinformersv1 "github.com/openshift/client-go/console/informers/externalversions/console/v1"
+	operatorinformersv1 "github.com/openshift/client-go/operator/informers/externalversions/operator/v1"
+	routesinformersv1 "github.com/openshift/client-go/route/informers/externalversions/route/v1"
+
+	// listers
+	operatorlistersv1 "github.com/openshift/client-go/operator/listers/operator/v1"
+	routeslisterv1 "github.com/openshift/client-go/route/listers/route/v1"
+
+	// clients
+	consoleclientv1 "github.com/openshift/client-go/console/clientset/versioned/typed/console/v1"
+	routeclientv1 "github.com/openshift/client-go/route/clientset/versioned/typed/route/v1"
+
+	// operator
+	"github.com/openshift/console-operator/pkg/api"
+	"github.com/openshift/console-operator/pkg/console/status"
+	routesub "github.com/openshift/console-operator/pkg/console/subresource/route"
+	"github.com/openshift/console-operator/pkg/console/subresource/util"
+)
+
+const (
+	controllerWorkQueueKey = "clidownloads-sync-work-queue-key"
+	controllerName         = "ConsoleCLIDownloadsSyncController"
+
+	ocCLIDownloadsCustomResourceName  = "oc-cli-downloads"
+	odoCLIDownloadsCustomResourceName = "odo-cli-downloads"
+)
+
+type CLIDownloadsSyncController struct {
+	// clients
+	operatorClient            v1helpers.OperatorClient
+	consoleCliDownloadsClient consoleclientv1.ConsoleCLIDownloadInterface
+	routeClient               routeclientv1.RoutesGetter
+	// listers
+	routesLister         routeslisterv1.RouteLister
+	operatorConfigLister operatorlistersv1.ConsoleLister
+	// events
+	cachesToSync []cache.InformerSynced
+	queue        workqueue.RateLimitingInterface
+	recorder     events.Recorder
+}
+
+func NewCLIDownloadsSyncController(
+	// clients
+	operatorClient v1helpers.OperatorClient,
+	cliDownloadsInterface consoleclientv1.ConsoleCLIDownloadInterface,
+	routeClient routeclientv1.RoutesGetter,
+	// informers
+	operatorConfigInformer operatorinformersv1.ConsoleInformer,
+	consoleCLIDownloadsInformers consoleinformersv1.ConsoleCLIDownloadInformer,
+	routesInformers routesinformersv1.RouteInformer,
+	// recorder
+	recorder events.Recorder,
+) *CLIDownloadsSyncController {
+
+	ctrl := &CLIDownloadsSyncController{
+		// clients
+		consoleCliDownloadsClient: cliDownloadsInterface,
+		routeClient:               routeClient,
+		// listers
+		operatorConfigLister: operatorConfigInformer.Lister(),
+		// events
+		recorder: recorder,
+		queue:    workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "ConsoleCliDownloadsSyncer"),
+	}
+
+	operatorClient.Informer().AddEventHandler(ctrl.newEventHandler())
+	operatorConfigInformer.Informer().AddEventHandler(ctrl.newEventHandler())
+	consoleCLIDownloadsInformers.Informer().AddEventHandler(ctrl.newEventHandler())
+	routesInformers.Informer().AddEventHandler(ctrl.newEventHandler())
```

Comment:
> So you can drop the routesLister now, we aren't using it.

shouldn't we keep it at least here ?

Reply:
So, you do need the `informer`, yes!
And the `client`.
But we don't need the `lister`.